wicketd/src/update_tracker.rs

```diff
@@ -2434,13 +2799,130 @@ impl<'a> SpComponentUpdateContext<'a> {
+                        move |_cx| async move {
+                            update_cx
+                                .reset_sp_component(SpComponent::ROT.const_as_str())
+                                .await
+                                .map_err(|error| {
+                                    SpComponentUpdateTerminalError::RotResetFailed {
+                                        error,
+                                    }
+                                })?;
+                            StepSuccess::new(()).into()
+                        },
+                    )
+                    .register();
+
+                registrar
+                    .new_step(
+                        SpComponentUpdateStepId::Resetting,
+                        "Waiting for RoT to boot".to_string(),
+                        move |_cx| async move {
+                            let (_, stage0next_error) = update_cx
+                                .wait_for_rot_boot_info(WAIT_FOR_BOOT_TIMEOUT)
+                                .await
+                                .map_err(|error| {
+                                    SpComponentUpdateTerminalError::GetRotBootInfoFailed { error }
+                                })?;
+
+                            // check that stage0next is valid before we try to set the component
+                            if let Some(error) = stage0next_error {
+                                return Err(SpComponentUpdateTerminalError::RotBootloaderError {
+                                    error: anyhow!(format!("{error:?}"))
+                                });
+                            }
+                            StepSuccess::new(()).into()
+                        },
+                    )
+                    .register();
+
+                // Actually set stage0 to use the new firmware
+                registrar
+                    .new_step(
+                        SpComponentUpdateStepId::SettingActiveBootSlot,
+                        format!("Setting {component_name} active slot to {firmware_slot}"),
+                        move |_cx| async move {
+                            update_cx
+                                .set_component_active_slot(
+                                    component_name,
+                                    firmware_slot,
+                                    true,
+                                )
+                                .await
+                                .map_err(|error| {
+                                    SpComponentUpdateTerminalError::SetRotActiveSlotFailed {
+                                        error,
+                                    }
+                                })?;
+                            StepSuccess::new(()).into()
+                        },
+                    )
+                    .register();
+
+                // Now reset (again) to boot into the new stage0
```

Comment:
Do we need to mutex this substep (or one of the other ones) (https://github.com/oxidecomputer/omicron/pull/5882#discussion_r1636913858)?

Reply:
I couldn't figure out how to place the mutex. We want to mutex at the rack level and it seemed like this gets spawned off per SP. This is probably my complete inexperience with Rust synchronization though.

Reply:
It's going to be a moderate-to-large mess no matter what, I think. How bad is landing this without it?

Reply:
For this particular set of stage0 updates it's pretty low risk. We're going to have some subsequent stage0 updates to add epochs to avoid roll backs so I think we could target it there.